### PR TITLE
Wbooks -- BookList screen components

### PR DESCRIPTION
**Pull request by lfmontesanto**
## Summary

- Book component with Cover, Title and Author
- BookList : FlatList of Book component 

## Screenshot
Screenshot_1581351256

## Trello Card

https://trello.com/c/edTylDPt/9-lista-de-libros

### CODE REVIEW

File: Wbooks/src/app/screens/library/components/book/styles.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import {StyleSheet} from 'react-native';
+
+const styles = StyleSheet.create({
+  author:{
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/styles.js]: `author: {`

File: Wbooks/src/app/screens/library/components/book/styles.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import {StyleSheet} from 'react-native';
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/styles.js]: You know

File: Wbooks/src/app/screens/library/components/book/styles.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+const styles = StyleSheet.create({
+  author:{
+    fontSize:15,
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/styles.js]: Insert spaces between key and values

File: Wbooks/src/app/screens/library/components/book/styles.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    marginLeft:30,
+    marginRight:19,
+  },
+  descriptionContainer: {
+    flexDirection:"column",
+    justifyContent:"flex-start",
+    marginTop: 15,
+  },
+  title: {
+    color:'#4A4A4A',
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/styles.js]: Move this to a colors const file, import and replace.

File: Wbooks/src/app/screens/library/components/book/styles.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    height: 90,
+    marginHorizontal:20,
+    marginBottom:10,
+  },
+  cover: {
+    width:40,
+    height:60,
+    marginVertical:15,
+    marginLeft:30,
+    marginRight:19,
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/styles.js]: Try with 20

File: Wbooks/src/app/screens/library/components/book/styles.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  },
+  descriptionContainer: {
+    flexDirection:"column",
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/styles.js]: Use `'` instead `"`

File: Wbooks/src/app/screens/library/components/book/styles.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    justifyContent:"flex-start",
+    marginTop: 15,
+  },
+  title: {
+    color:'#4A4A4A',
+    fontSize:17,
+    fontFamily:'SF UI Text Bold',
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/styles.js]: Move this to a fonts constants file, import it from there and replace.

File: Wbooks/src/app/screens/library/components/book/styles.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    fontWeight:'300',
+    letterSpacing:-0.24,
+    marginRight:26,
+    textAlign:"left",
+  },
+  bookContainer: {
+    display:'flex',
+    flexDirection:'row',
+    backgroundColor: '#FFFFFF',
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/styles.js]: Move this to a const colors file.

File: Wbooks/src/app/screens/library/components/book/styles.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
+import {StyleSheet} from 'react-native';
+
+const styles = StyleSheet.create({
+  author:{
+    fontSize:15,
+    lineHeight:18,
+    fontFamily:'SF UI Text Bold',
+    fontWeight:'300',
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/styles.js]: Delete this.

File: Wbooks/src/app/screens/library/components/book/styles.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+  cover: {
+    width:40,
+    height:60,
+    marginVertical:15,
+    marginLeft:30,
+    marginRight:19,
+  },
+  descriptionContainer: {
+    flexDirection:"column",
+    justifyContent:"flex-start",
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/styles.js]: Try deleting this

File: Wbooks/src/app/screens/library/components/book/styles.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+    fontWeight:"bold",
+    letterSpacing:-0.41,
+    lineHeight:20,
+    marginRight:26,
+    marginBottom:3,
+    textAlign:"left",
+  },
+});
+
+export default styles;
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/styles.js]: Insert tail line here.

File: Wbooks/src/app/screens/library/components/book/styles.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    marginTop: 15,
+  },
+  title: {
+    color:'#4A4A4A',
+    fontSize:17,
+    fontFamily:'SF UI Text Bold',
+    fontWeight:"bold",
+    letterSpacing:-0.41,
+    lineHeight:20,
+    marginRight:26,
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/styles.js]: Try with 25

File: Wbooks/src/app/screens/library/components/bookList/index.js [hunk 13]
```diff
@@ -0,0 +1,18 @@
+import React, {Component} from 'react'
+import {
+  View,
+} from 'react-native';
+import BookList from './layout.js';
+
+class BookListContainer extends Component {
+  render() {
+
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/bookList/index.js]: Delete this empty line

File: Wbooks/src/app/screens/library/components/bookList/index.js [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+    return(
+      <View>
+        <BookList bookList = {this.props.bookList} ></BookList>
+      </View>
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/bookList/index.js]: Ok, when you are returning just one component like in this case (just need to return BookList`) there is no need on wrap it on a View. The global rule is that the return statement can return just one component. On the other hand every View without styles means a component that is gonna be rendered without styles.

File: Wbooks/src/app/screens/library/components/bookList/index.js [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+    );
+  }
+}
+
+export default BookListContainer;
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/bookList/index.js]: Insert tail line here.

File: Wbooks/src/app/screens/library/components/bookList/layout.js [hunk 16]
```diff
@@ -0,0 +1,23 @@
...
+class BookList extends Component {
+  render() {
+    return (
+    <View style = {styles.container }>
+      <FlatList
+        data = {this.props.bookList}
+        renderItem={({item}) => <Book style = {styles.listItem} book = {item}/>}
+        keyExtractor={item => item.id.toString()}
+      />
+    </View>
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/bookList/layout.js]: Delete the View and apply the styles directly to the FlatList

File: Wbooks/src/app/screens/library/components/bookList/layout.js [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+import styles from './styles.js';
+import Book from '../book/index.js'
+
+class BookList extends Component {
+  render() {
+    return (
+    <View style = {styles.container }>
+      <FlatList
+        data = {this.props.bookList}
+        renderItem={({item}) => <Book style = {styles.listItem} book = {item}/>}
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/bookList/layout.js]: We don't use arrow functions inside the render, you should call it outside the render.

File: Wbooks/src/app/screens/library/components/bookList/layout.js [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+        keyExtractor={item => item.id.toString()}
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/bookList/layout.js]: Same here.

File: Wbooks/src/app/screens/library/components/bookList/styles.js [hunk 19]
```diff
@@ -0,0 +1,14 @@
+import {StyleSheet} from 'react-native';
+
+const styles = StyleSheet.create({
+   container: {
+    display:'flex',
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/bookList/styles.js]: Delete this.

File: Wbooks/src/app/screens/library/components/bookList/styles.js [hunk 20]
```diff
@@ -0,0 +1,14 @@
...
+    backgroundColor:'#EAF6FA',
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/bookList/styles.js]: Move color to const color file, import and replace them.

File: Wbooks/src/app/screens/library/components/bookList/styles.js [hunk 21]
```diff
@@ -0,0 +1,14 @@
...
+    paddingTop:16,
+    borderRadius:5,
+    shadowOffset: {width: 0,height: 1},
+    shadowColor:'#000000',
+  },
+});
+
+export default styles;
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/bookList/styles.js]: Insert tail line here.

File: Wbooks/index.js [hunk 22]
```diff
@@ -0,0 +1,9 @@
+/**
+ * @format
+ */
+
+import {AppRegistry} from 'react-native';
+import App from './src/app/App';
+import {name as appName} from './app.json';
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/index.js]: done

File: Wbooks/index.js [hunk 23]
```diff
@@ -0,0 +1,9 @@
+/**
+ * @format
+ */
+
+import {AppRegistry} from 'react-native';
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/index.js]: done

File: Wbooks/src/app/App.js [hunk 24]
```diff
@@ -0,0 +1,40 @@
...
+
+import React from 'react';
+import {
+  SafeAreaView,
+  StyleSheet,
+  ScrollView,
+  View,
+  Text,
+  StatusBar,
+} from 'react-native';
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/src/app/App.js]: done

File: Wbooks/src/app/App.js [hunk 25]
```diff
@@ -0,0 +1,40 @@
...
+
+import {
+  Header,
+  LearnMoreLinks,
+  Colors,
+  DebugInstructions,
+  ReloadInstructions,
+} from 'react-native/Libraries/NewAppScreen';
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/src/app/App.js]: done

File: Wbooks/src/app/App.js [hunk 26]
```diff
@@ -0,0 +1,40 @@
...
+
+
+import BookListContainer from './screens/library/components/bookList/index.js'
+
+const App: () => React$Node = () => {
+  let mockData = require('./mocklist.json');
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/src/app/App.js]: done

File: Wbooks/src/app/screens/library/components/bookList/layout.js (hunk 17 above)
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/bookList/layout.js]: done

File: Wbooks/src/app/screens/library/components/bookList/layout.js (hunk 18 above)
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/bookList/layout.js]: done

File: Wbooks/src/app/screens/library/components/bookList/styles.js (hunk 19 above)
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/bookList/styles.js]: done

File: Wbooks/src/app/screens/library/components/bookList/styles.js (hunk 20 above)
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/bookList/styles.js]: done

File: Wbooks/src/app/screens/library/components/bookList/styles.js (hunk 21 above)
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/bookList/styles.js]: done

File: Wbooks/src/app/App.js [hunk 27]
```diff
@@ -0,0 +1,40 @@
...
+  return (
+          <View>
+            <BookListContainer bookList = {mockData}/>
+          </View>
+
+  );
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/src/app/App.js]: done

File: Wbooks/src/app/screens/library/components/book/index.js [hunk 28]
```diff
@@ -0,0 +1,20 @@
...
+} from 'react-native';
+
+import Book from './layout.js';
+class BookContainer extends Component {
+  render() {
+    let book = this.props.book;
+    return(
+      <View>
+        <Book title = {book.title} author = {book.author} image_url = {book.image_url}></Book>
+      </View>
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/book/index.js]: done

File: Wbooks/src/app/screens/library/components/book/index.js [hunk 29]
```diff
@@ -0,0 +1,20 @@
+import React, {Component} from 'react'
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/book/index.js]: done

File: Wbooks/src/app/screens/library/components/book/index.js [hunk 30]
```diff
@@ -0,0 +1,20 @@
...
+import {
+  Image,
+  View,
+  Text,
+} from 'react-native';
+
+import Book from './layout.js';
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/book/index.js]: done

File: Wbooks/src/app/screens/library/components/book/index.js [hunk 31]
```diff
@@ -0,0 +1,20 @@
+import React, {Component} from 'react'
+import {
+  Image,
+  View,
+  Text,
+} from 'react-native';
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/book/index.js]: done

File: Wbooks/src/app/screens/library/components/book/index.js [hunk 32]
```diff
@@ -0,0 +1,20 @@
...
+    let book = this.props.book;
+    return(
+      <View>
+        <Book title = {book.title} author = {book.author} image_url = {book.image_url}></Book>
+      </View>
+    );
+  }
+}
+
+export default BookContainer;
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/book/index.js]: done

File: Wbooks/src/app/screens/library/components/book/layout.js [hunk 33]
```diff
@@ -0,0 +1,22 @@
+import React, {Component} from 'react'
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/book/layout.js]: done

File: Wbooks/src/app/screens/library/components/book/layout.js [hunk 34]
```diff
@@ -0,0 +1,22 @@
...
+import {
+  Image,
+  View,
+  Text,
+} from 'react-native';
+import styles from './styles.js';
+class Book extends Component {
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/book/layout.js]: done

File: Wbooks/src/app/screens/library/components/book/layout.js [hunk 35]
```diff
@@ -0,0 +1,22 @@
...
+  render() {
+    return (
+    <View style = {styles.bookContainer }>
+      <Image source={{uri: this.props.image_url}} style = {styles.cover}/>
+      <View style = {styles.descriptionContainer} >
+        <Text style = {styles.title}>{this.props.title}</Text>
+        <Text style = {styles.author}>{this.props.author}</Text>
+      </View>
+    </View>
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/book/layout.js]: done

File: Wbooks/src/app/screens/library/components/book/layout.js [hunk 36]
```diff
@@ -0,0 +1,22 @@
...
+    );
+  }
+};
+
+export default Book;
```
[the remark this answers is not in the document]
Comment (lfmontesanto) [reply Wbooks/src/app/screens/library/components/book/layout.js]: done

File: Wbooks/src/app/screens/library/components/bookList/layout.js [hunk 37]
```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react';
+import { FlatList } from 'react-native';
+
+import Book from '../book/index';
+
```
Comment (lfmontesanto) [inline Wbooks/src/app/screens/library/components/bookList/layout.js]: remove empty line right?

File: Wbooks/src/app/proptypes/proptypes.js [hunk 38]
```diff
@@ -0,0 +1,33 @@
...
+    PropTypes.shape({
+      author: PropTypes.string,
+      genre: PropTypes.string,
+      id: PropTypes.number,
+      image_url: PropTypes.string,
+      publisher: PropTypes.string,
+      title: PropTypes.string,
+      year: PropTypes.string
+    })
+  )
```
Comment (rodrigoWolox) [inline Wbooks/src/app/proptypes/proptypes.js]:
```
import { bool, number, string } from 'prop-types';

export const categoryModel = {
  id: number.isRequired,
  name: string.isRequired
};

export const articleModel = {
  id: number.isRequired,
  title: string,
  body: string,
  collapsed: bool
};

export const resultModel = {
  id: number.isRequired,
  sectionId: number.isRequired,
  title: string,
  sectionName: string
};
```
Implement this pattern to this proprTypes files

Comment (rodrigoWolox) [reply Wbooks/src/app/proptypes/proptypes.js]: As you can see we just import the necessary propTypes and declare each model by separate so we can reuse them even inside the same file.

File: Wbooks/src/app/screens/library/components/bookList/index.js [hunk 39]
```diff
@@ -0,0 +1,30 @@
...
+
+import styles from './styles';
+
+class BookListContainer extends Component {
+  renderItem = ({ item }) => <Book book={item} />;
+
+  keyExtractor = item => item.id.toString();
+
+  render() {
+    console.log(this.props.bookList);
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/bookList/index.js]: Delete this console

File: Wbooks/src/app/screens/library/components/bookList/index.js [hunk 40]
```diff
@@ -0,0 +1,30 @@
...
+    return (
+      <FlatList
+        style={styles.container}
+        data={this.props.bookList}
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/bookList/index.js]: You could destruct this value from props between render and its return
```
render() {
    const { bookList } = this.props;
    return (
```

File: Wbooks/src/app/screens/library/components/book/index.js [hunk 41]
```diff
@@ -0,0 +1,26 @@
...
+  return (
+    <View style={styles.bookContainer}>
+      <Image source={{ uri: imageUrl }} style={styles.cover} />
+      <View style={styles.descriptionContainer}>
+        <Text style={styles.title}>{title}</Text>
+        <Text style={styles.author}>{author}</Text>
+      </View>
+    </View>
+  );
+}
```
Comment (rodrigoWolox) [inline Wbooks/src/app/screens/library/components/book/index.js]: Nice!

**Pull request merged**